internal/sourceanalysis/go.go

```diff
@@ -17,6 +17,8 @@ import (
 	"golang.org/x/vuln/scan"
 )
 
+const goVersionEnv = "GOVERSION"
```

Comment:
is this a general env variable i.e. used by Go? or is it worth prefixing this with something like `OSV_SCANNER` to avoid conflict?

Reply:
Yeah, let's change this to OSV_SCANNER_GOVERSION for now, I don't think GOVERSION is actually interpreted by the go compiler in any way (e.g. if you set GOVERSION, it won't change the version that is actually being used, I think that's governed by GOROOT variable), but better to be safe here. 

This will just be a temporary fix, as we probably want this as something we can set in `osv-scanner.toml` eventually.

Reply:
Not really a general env variable. It's specifically used by govulncheck, and it will be parsed from "go env GOVERSION" if not defined. https://github.com/golang/vuln/blob/563994f0852c21fbdb2424841ebd8de389ff47ad/internal/scan/run.go#L78

Reply:
It can be changed to `OSV_SCANNER_GOVERSION` for safe when users want to specify a go version, but we still need to pass the `GOVERSION` to govulncheck.